doc/decisions/header_file_structure.md

```diff
@@ -12,31 +12,95 @@ The big problem is kdbprivate.h. It has two main problems:
 
 ## Constraints
 
+- Each library should have at least one separate header file:
+
+  Without this constraint, we would constantly include things that aren't needed.
+  This is inconvenient as it clutters the auto-suggestions of IDEs.
+  It also slows down the compile process, since the compiler has to parse everything that was included (see [also](https://lore.kernel.org/lkml/YdIfz+LMewetSaEB@gmail.com/T/)).
+
+- Non-public API and public API should be kept in separate files
+- Some non-public APIs need to be accessible for testing, such headers should not be packaged/installed
+
 ## Assumptions
 
+- There are different categories of "private":
+
+  1.  Some parts are truly private, i.e. shouldn't be used outside the library that defines them.
+      These things are only there because a library was split into multiple `.c` files.
+      This includes e.g. `splitNew` and `struct _Split`.
+
+      Symbols belonging to this category should not appear at all in the `symbols.map` file.
+
+  2.  Other things are truly private, but must be tested.
+      This includes e.g. most other `split*` functions or the `elektraKeyName*` functions.
+
+      Symbols belonging to this category should not appear in a public section of the `symbols.map` file.
```

Comment:
Do we need these symbols at all? In theory unit tests could simply include the .c files (with static functions).

Reply:
IMO including .c files is a bad idea.

1. It means that there will be a separate definition of the function in every file that includes the function. The performance aspect wouldn't matter for tests, but it would definitely break things like `static` variables (if we have those anywhere).
2. I think it's harder to define rules for when including `.c` files is okay and which files can be included where, than to just say "if you need to test a private function, make it non-static, add it to a `fortests` header and add it to the `fortests` section of the `symbols.map`".